llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -8599,6 +8655,8 @@ void LoopVectorizationPlanner::buildVPlansWithVPRecipes(ElementCount MinVF,
         VPlanTransforms::truncateToMinimalBitwidths(
             *Plan, CM.getMinimalBitwidths(), PSE.getSE()->getContext());
       VPlanTransforms::optimize(*Plan, *PSE.getSE());
+      if (CM.useVPIWithVPEVLVectorization())
+        VPlanTransforms::addExplicitVectorLength(*Plan);
```

Comment:
I tried it, explained it here long time ago https://reviews.llvm.org/D99750. It does not work because of some reason (some recipes are removed/transformed too early, so cannot put it later in the pipeline)

Reply:
still open to add TODO?